### PR TITLE
feat: add reason-language-server

### PR DESCRIPTION
**Pull request by ri7nz**
Just add reason-language-server

### CODE REVIEW

File: lua/lspconfig/server_configurations/reasonls.lua [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    settings = {},
+  },
+  docs = {
+    description = [[
+Reason language server
+
+**By default, reasonls doesn't have a `cmd` set.** This is because nvim-lspconfig does not make assumptions about your path.
+You have to install the language server manually.
+
+You can instsall reason language server from [reason-language-server](https://github.com/jaredly/reason-language-server) repository.
```
Comment (lithammer) [inline lua/lspconfig/server_configurations/reasonls.lua:17]: *install

**Comment by mjlbach** [comment]
`reasonls.lua` -> `reason_ls.lua`

**Comment by ri7nz** [comment]
@mjlbach  done for rename

File: lua/lspconfig/server_configurations/reason_ls.lua [hunk 2]
```diff
@@ -0,0 +1,25 @@
+local util = require 'lspconfig.util'
+
+return {
+  default_config = {
+    cmd = {},
```
Comment (mjlbach) [inline lua/lspconfig/server_configurations/reason_ls.lua:5]: Can we just put the binary name here anyways? That way people can add it to their path.

Comment (ri7nz) [reply lua/lspconfig/server_configurations/reason_ls.lua]: Yes, I have been add the binary name in `cmd`.

File: lua/lspconfig/server_configurations/reason_ls.lua [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    filetypes = { 'reason' },
+    root_dir = util.root_pattern('bsconfig.json', '.git'),
+    settings = {},
```
[the remark this answers is not in the document]
Comment (ri7nz) [reply lua/lspconfig/server_configurations/reason_ls.lua:8]: I'm not sure for put some settings here, so I put empty, but we can refer to this [documentation](https://github.com/jaredly/reason-language-server#configuration-settings)

[the remark this answers is not in the document]
Comment (ri7nz) [reply lua/lspconfig/server_configurations/reason_ls.lua:8]: I'm not sure for put some settings here, so I put empty, but we can refer to this [documentation](https://github.com/jaredly/reason-language-server#configuration-settings)

[the remark this answers is not in the document]
Comment (mjlbach) [reply lua/lspconfig/server_configurations/reason_ls.lua:8]: Just don't add the key then.

[the remark this answers is not in the document]
Comment (ri7nz) [reply lua/lspconfig/server_configurations/reason_ls.lua]: done

**Comment by mjlbach** [comment]
Lint is failing (one error is for hhvm, so you can ignore than or bundle fixing that lint in a separate commit).

**Comment by ri7nz** [comment]
@mjlbach  I have been format with stylua

**Pull request merged**